test/gtest/cat.hpp

```diff
@@ -0,0 +1,198 @@
+                 {64, 48, 7, 7}, {64, 48, 7, 7}, {64, 48, 7, 7}, {64, 48, 7, 7}}},
+            {1, {{65536, 16}, {65536, 351}}},                                         // dlrm
+            {2, {{512, 49, 1024}, {512, 49, 1024}}},                                  // gnmt
+            {1, {{256, 112, 14, 14}, {256, 288, 14, 14}, {256, 64, 14, 14},           // googlenet
+                 {256, 64, 14, 14}}},
+            {1, {{256, 256, 7, 7}, {256, 320, 7, 7}, {256, 128, 7, 7},
+                 {256, 128, 7, 7}}},
+            {0, {{2, 1024, 768}, {2, 1024, 768}, {2, 1024, 768}}},                    // GPT
+            {1, {{16, 192000, 1}, {16, 48000, 1}, {16, 12000, 1}, {16, 3000, 1},      // fasterrcnn
+                 {16, 780, 1}}},
+            {2, {{3990480, 1, 1}, {3990480, 1, 1}, {3990480, 1, 1}, {3990480, 1, 1}}},
+            {1, {{64, 320, 8, 8}, {64, 192, 8, 8}, {64, 768, 8, 8}}},                 // inceptionv3
+            {1, {{12277440, 1}, {12277440, 1}, {12277440, 1}, {12277440, 1}}},        // maskrcnn
+            {1, {{6, 182400, 4}, {6, 45600, 4}, {6, 11400, 4}, {6, 2850, 4},
+                 {6, 741, 4}}},
+            {0, {{255780}, {255780}, {255780}, {255780}, {255780}, {255780}}},
+            {1, {{4, 136800, 91}, {4, 34200, 91}, {4, 8550, 91}, {4, 2223, 91},       // retinanet
+                 {4, 630, 91}}},
+            {1, {{256, 232, 7, 7}, {256, 232, 7, 7}}},                                // shufflenetv2
+            {1, {{256, 116, 14, 14}, {256, 116, 14, 14}}},
+            {1, {{2, 192, 80, 80}, {2, 192, 80, 80}}},                                // yolor
+            {1, {{16, 320, 20, 20}, {16, 320, 20, 20}}},
+            {0, {{4096, 1312}, {4096, 1312}}},                                        // rnnt
+            {0, {{9600, 384}, {128, 384}}},                                           // roberta
+            {2, {{256, 81, 5776}, {256, 81, 2166}, {256, 81, 600}, {256, 81, 150},    // ssd
+                 {256, 81, 36}, {256, 81, 4}}},
+            {1, {{16, 256, 32, 32}, {16, 128, 32, 32}, {16, 64, 32, 32},              // stdc
+                 {16, 64, 32, 32}}},
+            {1, {{1536, 256, 13, 13}, {1536, 256, 13, 13}}},                          // squeezenet
+            {1, {{724, 64, 55, 55}, {724, 64, 55, 55}}}};
+    // clang-format on
+}
+
+inline int32_t SetTensorLayout(miopen::TensorDescriptor& desc)
+{
+    std::vector<std::size_t> lens = desc.GetLengths();
+    std::vector<int32_t> int32_t_lens(lens.begin(), lens.end());
+
+    // set the strides for the tensor
+    return SetTensorNd(&desc, int32_t_lens, desc.GetType());
+}
+
+template <typename T = float>
+struct CatTest : public ::testing::TestWithParam<CatTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle = get_handle();
+        cat_config    = GetParam();
+        std::mt19937 gen(0);
+        std::uniform_real_distribution<> d{-3, 3};
+        auto gen_value = [&](auto...) { return d(gen); };
```

Comment:
Please use generators from driver/random.hpp and test/random.hpp

Here is an example:
https://github.com/ROCm/MIOpen/pull/2583/commits/b90bf90ffd259e2c5a78efa183cb7d73615e1827#diff-bfa18307b5c43a529901ac79b9428ea97493d6d80182903fd8fa843aa3d0386c

Reply:
The testing time has been reduced by 4.5 times. Thank you.